### DISCUSSION TITLE
Can Dags in a DagBag be independent from other Dags ?

### DISCUSSION THREAD

**Discussion by huf-92** [apache/airflow#17344] [Q&A]
Airflow version 2.1.2

I am adding Dags in other locations than the Airflow Dags folder with

	***DagBag.py***

	import logging
	from airflow.models import DagBag

	# folders with dag definitions to add
	dags_dirs = ["random/path"]
	for dag_dir in dags_dirs:
		# add each DAG folder as DagBag
		dag_bag = DagBag(dag_dir)
		if dag_bag:
			for dag_id, dag in dag_bag.dags.items():
				globals()[dag_id] = dag
		else:
			logging.error("dag_bag error: %s", str(dag_bag))

which creates log files for each task defined in the added Dags.
Unfortunately there is no scheduler log file for each added Dag and therefore
error notification, for lets say import errors are only found in the scheduler
log of DagBag.py. Is there a way to force Airflow to create a scheduler log
file for each Dag added by a DagBag ? Furthermore if one Dag in the DagBag is
not valid non of the other Dags are working anymore. Is there a way to prevent
this ?

**Comment by XD-DENG** [comment]
May you pls share the use case for what you want to add multiple DAG folders? This is not an expected behaviour, so you have to expect a lot core code changes to support what you desire (if I understood it correctly).

One thing you may want to note is: you can actually have arbitrary number of sub-folders under Airflow's DAG_FOLDER. Maybe this is sufficient for what you need to do?

**Reply by huf-92** [reply]
@XD-DENG thanks for your reply. 
In my company we have several groups working on different topics. Each group should have their own Dag folder
with their own git repositories in it. The repositories contain the Dags and other stuff needed to execute the Dags.
Sub-folders under the Airflow's DAG_FOLDER is a solution, but Airflow needs an __init__.py in each sub-folder
to find the Dags of each group. In my opinion this is an inelegant solution. Preferable for me would be a DAG folder
for each group.
